modules/cats/src/smithy4s/catz/instances/HashInstances.scala

```diff
@@ -0,0 +1,26 @@
+package smithy4s.catz.instances
+
+import cats.{Eq, Hash}
+import cats.implicits.toContravariantOps
+import smithy4s.{~>, ByteArray, ShapeId, Timestamp}
+import smithy4s.schema.Primitive
+
+trait HashInstances {
+
+  implicit val byteArrayHash: Hash[ByteArray] =
+    new Hash[ByteArray] {
+      override def hash(x: ByteArray): Int = x.array.hashCode()
+
+      override def eqv(x: ByteArray, y: ByteArray): Boolean =
+        Eq[ByteArray].eqv(x, y)
+    }
+  implicit val documentHash: Hash[smithy4s.Document] =
+    Hash[String].contramap(_.toString)
+  implicit val shapeIdHash: Hash[ShapeId] = Hash[String].contramap(_.toString)
+  implicit val timeStampHash: Hash[Timestamp] =
+    Hash[Long].contramap(_.epochSecond)
+  implicit val primHashPf: ~>[Primitive, Hash] = Primitive.deriving[Hash]
```

Comment:
drop the `implicit` and also use `PolyFunction` instead of `~>`

Reply:
why prefer `PolyFunction`?

Reply:
To keep consistent with what he used for `Show`, and also because I'm thinking more and more about deleting `~>` to avoid conflicting with the cats one. Not 100% certain.